addon/-private/system/model/model.js

```diff
@@ -120,8 +120,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @type {Boolean}
     @readOnly
   */
-  hasDirtyAttributes: Ember.computed('currentState.isDirty', function() {
-    return this.get('currentState.isDirty');
+  hasDirtyAttributes: Ember.computed('currentState', function() {
+    return this.get('_internalModel.currentState.isDirty');
```

Comment:
@pangratz I switched the return back to `this.get('currentState.isDirty')`. Note the current failure on the branch. I think this should be returning `this.get('_internalModel.currentState.isDirty')`. In my previous question I was just wanting to verify with you that it was appropriate to refer to `_internalModel.currentState.isDirty`. That may have been confusing.